seps/1933-workload-identity-federation.md

```diff
@@ -0,0 +1,228 @@
+sequenceDiagram
+    participant Client as MCP Client
+    participant Auth as Authorization<br/>Server
+    participant OIDC as OIDC<br/>Discovery<br/>Endpoint
+    participant JWKS_URI as JWK Set<br/>Location
+    participant Server as MCP Server
+
+    Client->>Auth: 1. Request Access Token
+    Note over Auth: 2. Construct OpenID Provider Discovery Endpoint URL
+    Auth->>OIDC: 3. Request OpenID Provider Configuration Document
+    OIDC->>Auth: 4. Return OpenID Provider Configuration Document
+    Note over Auth: 5. Extract `jwks_uri`
+    Auth->>JWKS_URI: 6. Request JWK Set
+    JWKS_URI->>Auth: 7. Return JWK Set
+    Note over Auth: 8. Validate JWT<br/>Authorization Grant
+    Auth->>Client: 9. Return Access Token
+    Client->>Server: 10. MCP Server Access
+    Note over Server: 11. Validate Access Token
+    Server->>Client: 12. MCP Server Response
+```
+
+### Flow Steps
+
+1. **Request Access Token**: The MCP client requests an access token through a
+   POST request to the authorization server's token endpoint using the JWT Bearer
+   grant type as specified in
+   [RFC 7523 Section 2.1](https://datatracker.ietf.org/doc/html/rfc7523#section-2.1).
+   The request includes:
+   - `grant_type`: `urn:ietf:params:oauth:grant-type:jwt-bearer`
+   - `assertion`: The client's workload JWT credential
+   - `resource`: The canonical URI of the target MCP server (as defined in the
+     baseline MCP Authorization specification)
+
+2. **Construct OpenID Provider Discovery Endpoint URL**: The authorization server extracts the `iss` (issuer)
+   claim from the JWT and determines if it is a trusted issuer. If the issuer is
+   trusted, the authorization server constructs the OpenID Provider Discovery Endpoint URL from the issuer
+   claim as described in Section 4 of [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig).
+
+3. **Request OpenID Provider Configuration Document**: The authorization server requests the OpenID Provider Configuration Document
+   as described in Section 4.1 of [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationRequest).
+
+4. **Return OpenID Provider Configuration Document** The OpenID Provider Configuration Document is returned as described in Section 4.2 of [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse).
+
+5. **Extract `jwks_uri`**: The authorization server validates the OpenID Provider Configuration Document as
+   described in Section 4.3 of [OpenID Connect Discovery 1.0](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationValidation) and extracts the `jwks_uri` configuration information.
+
+6. **Request JWK Set**: The authorization server retrieves the
+   [RFC7517](https://datatracker.ietf.org/doc/html/rfc7517) JWK set
+   containing the issuer's public keys from the TLS protected location specified by
```

Comment:
Assuming "TLS protected location" is basically a HTTPS endpoint?

Reply:
Correct - I think I borrowed language from another draft - happy to clarify that in practice this is an HTTPS endpoint.